packages-exp/auth-types-exp/index.d.ts

```diff
@@ -92,114 +102,324 @@ export interface ParsedToken {
+   * popup/redirect operations provided the specified providers support localization with the 
+   * language code specified.
+   */
   languageCode: string | null;
+  /**
+   * The current Auth instance's tenant ID. This is a readable/writable property. When you set 
+   * the tenant ID of an Auth instance, all future sign-in/sign-up operations will pass this 
+   * tenant ID and sign in or sign up users to the specified tenant project. When set to null, 
+   * users are signed in to the parent project. By default, this is set to null.
+   * 
+   * @example
+   * ```
+   * // Set the tenant ID on Auth instance.
+   * auth.tenantId = ‘TENANT_PROJECT_ID’;
+   * 
+   * // All future sign-in request now include tenant ID.
+   * const result = await signInWithEmailAndPassword(auth, email, password);
+   * // result.user.tenantId should be ‘TENANT_PROJECT_ID’.
+   * ```
+   */
   tenantId: string | null;
+  /**
+   * The current Auth instance's settings. This is used to edit/read configuration related options 
+   * like app verification mode for phone authentication.
+   */
   readonly settings: AuthSettings;
+  /**
+   * Adds an observer for changes to the user's sign-in state.
+   * 
+   * Prior to 4.0.0, this triggered the observer when users were signed in, signed out, or when 
+   * the user's ID token changed in situations such as token expiry or password change. After 
+   * 4.0.0, the observer is only triggered on sign-in or sign-out.
+   * 
+   * To keep the old behavior, see {@link Auth.onIdTokenChanged}.
+   * 
+   * @param nextOrObserver - callback triggere on change.
+   * @param error - callback triggered on error.
+   * @param completed - callback triggered when observer is removed.
+   */
   onAuthStateChanged(
     nextOrObserver: NextOrObserver<User>,
     error?: ErrorFn,
     completed?: CompleteFn
   ): Unsubscribe;
+  /**
+   * 
+   * Adds an observer for changes to the signed-in user's ID token, which includes sign-in, 
+   * sign-out, and token refresh events. This method has the same behavior as 
+   * {@link Auth.onAuthStateChanged} had prior to 4.0.0.
+   * 
+   * @param nextOrObserver - callback triggere on change.
+   * @param error - callback triggered on error.
+   * @param completed - callback triggered when observer is removed.
+   */
   onIdTokenChanged(
     nextOrObserver: NextOrObserver<User>,
     error?: ErrorFn,
     completed?: CompleteFn
   ): Unsubscribe;
+  /** The currently signed-in user (or null). */
   readonly currentUser: User | null;
+  /**
+   * Asynchronously sets the provided user as `currentUser` on the current Auth instance. A new 
+   * instance copy of the user provided will be made and set as currentUser.
+   * 
+   * This will trigger {@link Auth.onAuthStateChanged} and {@link Auth.onIdTokenChanged} listeners 
+   * like other sign in methods.
+   * 
+   * The operation fails with an error if the user to be updated belongs to a different Firebase 
```

Comment:
We also don't currently check this. Filed b/170854629

Reply:
thanks